include/aikido/planner/vectorfield/VectorFieldUtil.hpp

```diff
@@ -41,47 +34,83 @@ class DesiredTwistFunction : public dart::optimizer::Function
-/// and linear velocity
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in]  _bodyNode Body node of the end-effector
-/// \param[in] _optimizationTolerance Callback of vector field calculation
-/// \param[in] _timestep How long will the computed joint velocities be executed
-/// \param[in] _padding Padding for joint limits
-/// \param[out] _jointVelocity Calculated joint velocities
+/// \param[out] jointVelocity Calculated joint velocities.
+/// \param[in] desiredTwist Desired twist, which consists of angular velocity
+/// and linear velocity.
+/// \param[in] stateSpace MetaSkeleton state space.
+/// \param[in] bodyNode Body node of the end-effector.
+/// \param[in] jointLimitPadding If less then this distance to joint
+/// limit, velocity is bounded in that direction to 0.
+/// \param[in] jointVelocityLowerLimits Joint velocity lower bounds.
+/// \param[in] jointVelocityUpperLimits Joint velocity upper bounds.
+/// \param[in] jointVelocityLimited Whether joint velocity bounds are
+/// considered in optimization.
+/// \param[in] stepSize Step size in second.
+/// \return Whether a joint velocity is found
 bool computeJointVelocityFromTwist(
-    const Eigen::Vector6d& _desiredTwist,
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
-    const dart::dynamics::BodyNodePtr _bodyNode,
-    double _optimizationTolerance,
-    double _timestep,
-    double _padding,
-    Eigen::VectorXd& _jointVelocity);
+    Eigen::VectorXd& jointVelocity,
+    const Eigen::Vector6d& desiredTwist,
+    const aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+    const dart::dynamics::BodyNodePtr bodyNode,
+    double jointLimitPadding,
+    const Eigen::VectorXd& jointVelocityLowerLimits,
+    const Eigen::VectorXd& jointVelocityUpperLimits,
+    bool jointVelocityLimited,
+    double stepSize);
+
+/// Compute the twist in global coordinate that corresponds to the gradient of
+/// the geodesic distance between two transforms.
+///
+/// \param[in] currentTrans Current transformation.
+/// \param[in] goalTrans Goal transformation.
+Eigen::Vector6d computeGeodesicTwist(
+    const Eigen::Isometry3d& currentTrans, const Eigen::Isometry3d& goalTrans);
+
+/// Compute the error in gloabl coordinate between two transforms.
+///
+/// \param[in] currentTrans Current transformation.
+/// \param[in] goalTrans Goal transformation.
+Eigen::Vector4d computeGeodesicError(
+    const Eigen::Isometry3d& currentTrans, const Eigen::Isometry3d& goalTrans);
+
+/// Compute the geodesic distance between two transforms.
+/// gd = norm( relative translation + r * axis-angle error )
+///
+/// \param[in] currentTrans Current transformation.
+/// \param[in] goalTrans Goal transformation.
+/// \param[in] r In units of meters/radians converts radians to meters.
+double computeGeodesicDistance(
+    const Eigen::Isometry3d& currentTrans,
+    const Eigen::Isometry3d& goalTrans,
+    double r = 1.0);
```

Comment:
I don't think `r` should have a default. There's no reason to think that `1` is a good value for this - in fact, this is likely far too large for HERB because his entire workspace is only ~2 m wide.